packages/matter-node.js/src/matter/MatterDevice.ts

```diff
@@ -60,8 +69,61 @@ export class MatterDevice {
         return this;
     }
 
-    start() {
-        this.broadcasters.forEach(broadcaster => broadcaster.announce());
+    async start() {
+        /*for (const broadcaster of this.broadcasters) {
+            broadcaster.setCommissionMode(0, this.deviceName, this.deviceType, this.vendorId, this.productId, this.discriminator);
+            broadcaster.announce();
+        }*/
+        this.startAnnouncement();
+    }
+
+    startAnnouncement() {
```

Comment:
Shouldn't this announce logic live in the library somewhere rather than coded directly into the sample Device application?

Reply:
MatterDevice is the "device node" logic level right now. (Example device is Device.ts). That will move also with high level api anyway.